src/coreclr/gc/gc.cpp

```diff
@@ -22829,6 +22977,23 @@ void gc_heap::mark_phase (int condemned_gen_number, BOOL mark_only_p)
         else
             mark_list_end = &mark_list [0];
         mark_list_index = &mark_list [0];
+
+#if defined(USE_REGIONS) && defined(MULTIPLE_HEAPS)
+        if (g_mark_list_piece != nullptr)
+        {
+            // two arrays with alloc_count entries per heap
+            mark_list_piece_start = &g_mark_list_piece[heap_number * 2 * g_mark_list_piece_size];
+            mark_list_piece_end = &mark_list_piece_start[g_mark_list_piece_size];
+        }
+        else
+        {
+            // disable use of mark list altogether
+            mark_list_piece_start = nullptr;
+            mark_list_piece_end = nullptr;
+            mark_list_end = &mark_list[0];
```

Comment:
this would mean that you will set `mark_list_overflow` to true in `sort_mark_list` even though you don't need to grow, wouldn't it?

Reply:
Ah, right, great catch - thanks! I will test for the case of g_mark_list_piece being null at the start of sort_mark_list and treat it the same as a gen 2.